tests/dialects/test_tsql.py

```diff
@@ -2090,3 +2090,19 @@ def test_next_value_for(self):
                 "oracle": "SELECT NEXT VALUE FOR db.schema.sequence_name",
             },
         )
+
+    def test_datetrunc(self):
+        self.validate_all(
+            "SELECT DATETRUNC(month, '2021-12-08 11:30:15.1234567')",
+            write={
+                "duckdb": "SELECT DATE_TRUNC('MONTH', CAST('2021-12-08 11:30:15.1234567' AS TIMESTAMP))"
+            },
+        )
+        self.validate_all(
+            "SELECT DATETRUNC(year, DATEFROMPARTS(2010, 12, 31))",
+            write={"duckdb": "SELECT DATE_TRUNC('YEAR', MAKE_DATE(2010, 12, 31))"},
+        )
```

Comment:
Try not to inline the `write` dict, prefer to go with the "expanded" version similar to the first case i.e:

```Python3
        self.validate_all(
            "SELECT DATETRUNC(year, DATEFROMPARTS(2010, 12, 31))",
            write={
                "duckdb": "SELECT DATE_TRUNC('YEAR', MAKE_DATE(2010, 12, 31))",
            },
        }

``` 

Also, the tests don't have to be "real", imo it'd be cleaner to have something like `"SELECT DATETRUNC(year, foo)"` to showcase the non-literal case